junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/TestInstancePostProcessorAndPreDestroyCallbackTests.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2015-2019 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.extension;
+
+import static java.util.Arrays.asList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestInstancePostProcessor;
+import org.junit.jupiter.api.extension.TestInstancePreDestroyCallback;
+import org.junit.jupiter.engine.AbstractJupiterTestEngineTests;
+import org.junit.jupiter.engine.JupiterTestEngine;
+
+/**
+ * Integration tests that verify support for {@link TestInstancePostProcessor}
+ * and {@link TestInstancePreDestroyCallback} in the {@link JupiterTestEngine}.
+ *
+ * @since 5.6
+ */
+class TestInstancePostProcessorAndPreDestroyCallbackTests extends AbstractJupiterTestEngineTests {
+
+	private static final List<String> callSequence = new ArrayList<>();
+
+	@Test
+	void beforeAllAndAfterAllCallbacks() {
+		// @formatter:off
+        assertBeforeAllAndAfterAllCallbacks(TopLevelTestCase.class,
```

Comment:
Watch out, the spaces are striking back! 😛

Reply:
agh, thought I'd got them all, do you have IntelliJ IDE formatting rules I can import to make it easier to catch these?

Reply:
looks like space were creeping in over tabs, added tabs back in.

Reply:
> agh, thought I'd got them all, do you have IntelliJ IDE formatting rules I can import to make it easier to catch these?

Yes, see [here](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#formatting) for details about how to use them with IntelliJ ;)